### ISSUE TITLE
<ebay-toggle-button-group> Marko type check complaining about `on-change` for `ebay-toggle-button-group`

### ISSUE THREAD

**Issue by malughanshyam** [eBay/ebayui-core#2064]
# Bug Report

## Description

Marko type check complaining about `on-change` for `ebay-toggle-button-group`

```
Object literal may only specify known properties, and '"on-change"' does not exist in type 'Directives & Input'.script(2353)
```

I see another marko-type-check problem.
Let me know if you want me to open another issue. Or if this can be handled on application side.

```

src/components/page-unified/index.marko:75:33 - error TS2322

  73 |                                 rejectText=(content.getText("productType.modal.cancel") as string)
  74 |                                 open=state.openPageModalDialog
> 75 |                                 on-reject("handleStartOver")
     |                                 ^^^^^^^^^ 
Type '(ev: any) => void' is not assignable to type '() => void'.
Target signature provides too few arguments. Expected 1 or more, but got 0.
  76 |                                 on-confirm("handleContinueBuilding")
  77 |                             >
  78 |                                 <@header>

node_modules/@ebay/ebayui-core/dist/components/ebay-confirm-dialog/index.d.marko:7:5 - message TS6500

   5 |     "confirm-text"?: string;
   6 |     "on-open"?: () => void;
>  7 |     "on-reject"?: () => void;
     |     ^^^^^^^^^^^ The expected type comes from property 'on-reject' which is declared here on type 'Directives & Input'
```

## Screenshots
![image](https://github.com/eBay/ebayui-core/assets/5460329/9d507515-e574-40fe-bc95-72535318cbb7)

**Comment by malughanshyam** [comment]
Hi @LuLaValva :
I see another marko-type-check problem.
Let me know if you want me to open another issue. Or if this can be handled on application side.

src/components/page-unified/index.marko:75:33 - error TS2322

  73 |                                 rejectText=(content.getText("productType.modal.cancel") as string)
  74 |                                 open=state.openPageModalDialog
> 75 |                                 on-reject("handleStartOver")
     |                                 ^^^^^^^^^ 
Type '(ev: any) => void' is not assignable to type '() => void'.
Target signature provides too few arguments. Expected 1 or more, but got 0.
  76 |                                 on-confirm("handleContinueBuilding")
  77 |                             >
  78 |                                 <@header>

node_modules/@ebay/ebayui-core/dist/components/ebay-confirm-dialog/index.d.marko:7:5 - message TS6500

   5 |     "confirm-text"?: string;
   6 |     "on-open"?: () => void;
>  7 |     "on-reject"?: () => void;
     |     ^^^^^^^^^^^ The expected type comes from property 'on-reject' which is declared here on type 'Directives & Input'

**Comment by malughanshyam** [comment]
Hi @LuLaValva : 
Another issue: 
`ebay-section-notice`

```

   9 |     data-testid="notice-banner-impacted-listings"
  10 | >
> 11 |     <@title>
     |      ^^^^^^ Object literal may only specify known properties, and '[/*@title*/Symbol.iterator]' does not exist in type 'Title & { as?: Renderable | undefined; }'.
  12 |         <div data-testid="notice-banner-impacted-listings-title">
  13 |             ${content.getText("listings.impactedListings.notice.title", {
  14 |                 x: String(listingsCount)

node_modules/@ebay/ebayui-core/dist/components/components/ebay-notice-base/component.d.ts:17:5 - message TS6500

  15 |     footer?: Marko.Renderable;
  16 |     "a11y-dismiss-text"?: string;
> 17 |     title?: Marko.Input<"title"> & {
     |     ^^^^^ The expected type comes from property 'title' which is declared here on type 'Directives & Input'
  18 |         as?: Marko.Renderable;
  19 |     };
  20 |     cta?: Marko.Input<"a"> & {
```

**Comment by LuLaValva** [comment]
Thanks for the extra issues @malughanshyam.

The section notice issue should be fixed with #2070, but I will look into the event handler.

I will explore the idea of including an additional `...any[]` argument in _all_ event handlers, so these kinds of issues won't be blocking for `mtc`. This change might also be the only way to allow additional parameters passed in via Marko files.

**Comment by malughanshyam** [comment]
Hi @LuLaValva : 
We have a macro defined with `name="message`

```
<macro|{ count, localizedCount, iconType, singularKey, pluralKey }| name="message">
    <div..>
```

Marko type check has issues with it.. Please advise. Should I create a new issue for this? 

```
src/components/section-manage-ad-groups/index.marko:77:18 - error TS2552

  75 |             </else-if>
  76 |             <else-if(input.messageType === AdGroupMessageEnum.NO_GROUP)>
> 77 |                 <message
     |                  ^^^^^^^ Cannot find name 'message'. Did you mean 'onmessage'?
node_modules/typescript/lib/lib.dom.d.ts:27888:13 - message TS2728
> 27888 | declare var onmessage: ((this: Window, ev: MessageEvent) => any) | null;
        |             ^^^^^^^^^ 'onmessage' is declared here.
```
